lib/Sema/TypeCheckProtocolInference.cpp

```diff
@@ -546,8 +546,8 @@ static Type getWitnessTypeForMatching(NormalProtocolConformance *conformance,
 
 /// Remove the 'self' type from the given type, if it's a method type.
 static Type removeSelfParam(ValueDecl *value, Type type) {
-  if (auto func = dyn_cast<AbstractFunctionDecl>(value)) {
-    if (func->getDeclContext()->isTypeContext())
+  if (isa<AbstractFunctionDecl>(value) || isa<EnumElementDecl>(value)) {
+    if (value->getDeclContext()->isTypeContext())
```

Comment:
You can use `value->hasCurriedSelf()` here :)

Reply:
Nice! Updated to use that now :)